spec/models/message_spec.rb

```diff
@@ -32,4 +32,18 @@
     end
   end
 
+  describe "assign_attributesメソッド" do
+    it "フォームオブジェクトにパラメータの値が反映される" do
+      message.assign_attributes(body: "message", pictures_attributes: )
```

Comment:
下記のように書いてpictureを追加できないですかね？
```suggestion
      message.body = "message"
      message.pictures << build(:picture)
```

Reply:
すみません、少し理解できなかったので質問させてください。

私のイメージでは`assign_attributes`にパラメータを入れ、
期待した返り値が得られるのを確認できれば
テストとしてOKかと思ったのですが、そうではないのでしょうか？

メソッドのテストがまだ理解できていないので、
どういったテストを書くのが良いか教えていただけないでしょうか。

Reply:
> 私のイメージではassign_attributesにパラメータを入れ、
期待した返り値が得られるのを確認できれば
テストとしてOKかと思ったのですが、そうではないのでしょうか？

あー、ここってassign_attributesのテストか、失礼しました。
ただ、モデルのテストなのでassign_attributesのテストはspec/forms配下に書くべきではないですかね。

message_formでのpictures_attributesの指定方法ですが、下記のような書き方になるかと思います。
```
message.assign_attributes(body: "message", pictures_attributes: [{ picture: File.new("#{Rails.root}/spec/factories/images/rails.png") }])
```

> メソッドのテストがまだ理解できていないので、どういったテストを書くのが良いか教えていただけないでしょうか。

request specでformのassign_attributesもsaveも通過するんで、僕は書かなくてもいいかと思います。同じテストを2回することになるので。複雑な処理をしていて細かいパターンを調べたいのであれば書く必要あるかと思いますが。

Reply:
`pictures_attributes`の回答ありがとうございます。
普通にファイルを作成すれば良いだけなのですね。

> request specでformのassign_attributesもsaveも通過するんで、僕は書かなくてもいいかと思います。同じテストを2回することになるので。複雑な処理をしていて細かいパターンを調べたいのであれば書く必要あるかと思いますが。

コントローラ側で動いていて、それほど複雑な処理でないなら
モデルにテストを書かなくてもいいのですね。それではこの部分は削除しておきます。